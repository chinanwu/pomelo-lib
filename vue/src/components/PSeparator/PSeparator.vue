<template>
  <div class="PSeparator" role="presentation"></div>
</template>

<script>
export default {
  name: "PSeparator",
  props: {
    vertical: {
      type: Boolean,
      default: false
    }
  }
};
</script>

<style lang="less" scoped>
.PSeparator {
  margin: 3rem;
  border-bottom: 1px solid #f2f2f2;
  width: 60%;
}

@media only screen and (max-width: 375px) {
  .PSeparator {
    width: 80%;
  }
}
</style>
