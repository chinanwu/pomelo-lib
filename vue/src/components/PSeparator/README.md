# PSeparator

A component used for separating bodies of content. Really is just a line.

[Click here to see a demo of these buttons in action](https://chinanwu.github.io/pomelo-lib/vue/index.html#pseparator)

## Example
```html
<PSeparator></PSeparator>
```
Alternatively, you can use the hyphenated component name.
```html
<p-separator></p-separator>
```

## Best practices

- TBA ?

## See an issue?
- Please create an issue [here](https://github.com/chinanwu/pomelo-lib/issues). Indicate the component and the issue, thank you! 
- Alternatively, if you fix the issues, you can create a [pull request](https://github.com/chinanwu/pomelo-lib/pulls)! 
