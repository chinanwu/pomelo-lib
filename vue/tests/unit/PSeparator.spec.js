import { shallowMount } from "@vue/test-utils";

import PSeparator from "@/components/PSeparator/PSeparator";

describe("PSeparator", () => {
  describe("rendering", () => {
    it("renders PSeparator", () => {
      const wrapper = shallowMount(PSeparator);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
