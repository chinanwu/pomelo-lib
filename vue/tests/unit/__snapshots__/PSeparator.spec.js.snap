// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PSeparator rendering renders PSeparator 1`] = `
<div
  class="PSeparator"
  role="presentation"
/>
`;
